BenchmarkReadMessage/FundingLocked failure

I don't think we run the benchmarks in the test framework or this would have been caught. This happens since there is now a scid alias in funding_locked and `createExtraData` is called which populates the extra bytes with random data and can produce a non-canonical tlv stream:
```
--- FAIL: BenchmarkReadMessage/FundingLocked
    message_test.go:244: 
        	Error Trace:	message_test.go:244
        	            				benchmark.go:193
        	            				benchmark.go:233
        	            				asm_amd64.s:1571
        	Error:      	Received unexpected error:
        	            	tlv stream is not canonical
        	Test:       	BenchmarkReadMessage/FundingLocked
        	Messages:   	unable to read msg
```

--- a/Makefile
+++ b/Makefile
@@ -196,6 +196,10 @@ unit-race:
 	@$(call print, "Running unit race tests.")
 	env CGO_ENABLED=1 GORACE="history_size=7 halt_on_errors=1" $(UNIT_RACE)
 
+unit-bench: $(BTCD_BIN)
+	@$(call print, "Running benchmark tests.")
+	$(UNIT_BENCH)
+
 # =============
 # FLAKE HUNTING
 # =============

--- a/docs/release-notes/release-notes-0.17.0.md
+++ b/docs/release-notes/release-notes-0.17.0.md
@@ -17,6 +17,9 @@
 
 * [Ensure that both the byte and string form of a TXID is populated in the 
   lnrpc.Outpoint message](https://github.com/lightningnetwork/lnd/pull/7624). 
+  
+* [Fix Benchmark Test (BenchmarkReadMessage/Channel_Ready) in the lnwire 
+package](https://github.com/lightningnetwork/lnd/pull/7356)
 
 ## RPC
 
@@ -49,3 +52,4 @@ unlock or create.
 * Erik Arvstedt
 * hieblmi
 * Jordi Montes
+* ziggie1984

--- a/lnwire/message_test.go
+++ b/lnwire/message_test.go
@@ -452,8 +452,20 @@ func newMsgChannelReady(t testing.TB, r io.Reader) *lnwire.ChannelReady {
 
 	pubKey := randPubKey(t)
 
-	msg := lnwire.NewChannelReady(lnwire.ChannelID(c), pubKey)
-	msg.ExtraData = createExtraData(t, r)
+	// When testing the ChannelReady msg type in the WriteMessage
+	// function we need to populate the alias here to test the encoding
+	// of the TLV stream.
+	aliasScid := lnwire.NewShortChanIDFromInt(rand.Uint64())
+	msg := &lnwire.ChannelReady{
+		ChanID:                 lnwire.ChannelID(c),
+		NextPerCommitmentPoint: pubKey,
+		AliasScid:              &aliasScid,
+		ExtraData:              make([]byte, 0),
+	}
+
+	// We do not include the TLV record (aliasScid) into the ExtraData
+	// because when the msg is encoded the ExtraData is overwritten
+	// with the current aliasScid value.
 
 	return msg
 }

--- a/make/testing_flags.mk
+++ b/make/testing_flags.mk
@@ -100,12 +100,16 @@ ifeq ($(UNIT_TARGETED), yes)
 UNIT := $(GOTEST) -tags="$(DEV_TAGS) $(LOG_TAGS)" $(TEST_FLAGS) $(UNITPKG)
 UNIT_DEBUG := $(GOTEST) -v -tags="$(DEV_TAGS) $(LOG_TAGS)" $(TEST_FLAGS) $(UNITPKG)
 UNIT_RACE := $(GOTEST) -tags="$(DEV_TAGS) $(LOG_TAGS) lowscrypt" $(TEST_FLAGS) -race $(UNITPKG)
+# NONE is a special value which selects no other tests but only executes the benchmark tests here.
+UNIT_BENCH := $(GOTEST) -tags="$(DEV_TAGS) $(LOG_TAGS)" -test.bench=. -test.run=NONE $(UNITPKG)
 endif
 
 ifeq ($(UNIT_TARGETED), no)
 UNIT := $(GOLIST) | $(XARGS) env $(GOTEST) -tags="$(DEV_TAGS) $(LOG_TAGS)" $(TEST_FLAGS)
 UNIT_DEBUG := $(GOLIST) | $(XARGS) env $(GOTEST) -v -tags="$(DEV_TAGS) $(LOG_TAGS)" $(TEST_FLAGS)
 UNIT_RACE := $(UNIT) -race
+# NONE is a special value which selects no other tests but only executes the benchmark tests here.
+UNIT_BENCH := $(GOLIST) | $(XARGS) env $(GOTEST) -tags="$(DEV_TAGS) $(LOG_TAGS)" -test.bench=. -test.run=NONE
 endif
 
 

